### PR TITLE
GEODE-3415: Fix SSL certificate chain support

### PR DESCRIPTION
**Pull request by mmartell**
Co-authored-by: Ivan Godwin <igodwin@pivotal.io>
Co-authored-by: Jake Barrett <jbarrett@pivotal.io>

### CODE REVIEW

File: cryptoimpl/SSLImpl.cpp [hunk 1]
```diff
@@ -56,16 +60,26 @@ SSLImpl::SSLImpl(ACE_HANDLE sock, const char *pubkeyfile,
...
     if (strlen(password) > 0) {
       SSL_CTX_set_default_passwd_cb(sslctx->context(), pem_passwd_cb);
       SSL_CTX_set_default_passwd_cb_userdata(sslctx->context(),
                                              const_cast<char *>(password));
     }
 
-    sslctx->private_key(privkeyfile);
-    sslctx->certificate(privkeyfile);
+    if (sslctx->private_key(privkeyfile) != 0) {
+      throw Exception("Failed to read private key.");
```
Comment (pivotal-jbarrett) [inline cryptoimpl/SSLImpl.cpp]: There ought to be a better Exception class to throw here.

Comment (mmartell) [reply cryptoimpl/SSLImpl.cpp]: Added new SSLException type and instantiated with appropriate exception strings.

File: cppcache/include/geode/ExceptionTypes.hpp [hunk 2]
```diff
@@ -812,6 +812,18 @@ class APACHE_GEODE_EXPORT PutAllPartialResultException : public Exception {
   }
 };
 
+/**
+ *@brief Thrown when an error is encountered during an SSL operation.
+ **/
+class APACHE_GEODE_EXPORT SSLException : public Exception {
+ public:
+  using Exception::Exception;
+  ~SSLException() noexcept override {}
```
Comment (pivotal-jbarrett) [inline cppcache/include/geode/ExceptionTypes.hpp]: `SslException` would be preferable for camel casing.

Comment (echobravopapa) [reply cppcache/include/geode/ExceptionTypes.hpp]: except that `SSL` is an acronym... didn't think we liked to camel them...

Comment (pivotal-jbarrett) [reply cppcache/include/geode/ExceptionTypes.hpp]: https://google.github.io/styleguide/cppguide.html#General_Naming_Rules
> For some symbols, this style guide recommends names to start with a capital letter and to have a capital letter for each new word (a.k.a. "Camel Case" or "Pascal case"). When abbreviations or acronyms appear in such names, prefer to capitalize the abbreviations or acronyms as single words (i.e `StartRpc()`, not `StartRPC()`).

Comment (echobravopapa) [reply cppcache/include/geode/ExceptionTypes.hpp]: fair nuf!

Comment (mmartell) [reply cppcache/include/geode/ExceptionTypes.hpp]: Done

**Pull request merged**